fortio_main.go

```diff
@@ -150,12 +156,35 @@ func fortioLoad() {
 		res, err = fhttp.RunHTTPTest(&o)
 	}
 	if err != nil {
-		fmt.Printf("Aborting because %v\n", err)
+		fmt.Fprintf(out, "Aborting because %v\n", err)
 		os.Exit(1)
 	}
-	fmt.Printf("All done %d calls (plus %d warmup) %.3f ms avg, %.1f qps\n",
+	fmt.Fprintf(out, "All done %d calls (plus %d warmup) %.3f ms avg, %.1f qps\n",
 		res.Result().DurationHistogram.Count,
 		*numThreadsFlag,
-		1000.*res.Result().DurationHistogram.Avg(),
+		1000.*res.Result().DurationHistogram.Avg,
 		res.Result().ActualQPS)
+	jsonFileName := *jsonFlag
+	if len(jsonFileName) > 0 {
+		j, err := json.MarshalIndent(res, "", "  ")
+		if err != nil {
+			log.Fatalf("Unable to json serialize result: %v", err)
+		}
+		var f *os.File
+		if jsonFileName == "-" {
+			f = os.Stdout
+			jsonFileName = "stdout"
+		} else {
+			f, err = os.Create(jsonFileName)
+			if err != nil {
+				log.Fatalf("Unable to create %s: %v", jsonFileName, err)
+			}
+		}
+		n, err := f.Write(j)
+		if err != nil {
+			log.Fatalf("Unable to write json to %s: %v", jsonFileName, err)
+		}
+		f.Write([]byte{'\n'})
+		fmt.Fprintf(out, "Succesfully wrote %d bytes of Json data to %s\n", n, jsonFileName)
+	}
```

Comment:
Shouldn't the json file be closed with `f.Close()`?
P.S. I understand that there is a preference in Go to avoid deferring the file close calls to be able to catch its return error.

Reply:
it adds quite a lot of rather ugly code and unnecessary code (the program will exit and close open fds) but you're right, changing/adding